mcda/electre/concordance.py

```diff
@@ -1,617 +1,660 @@
+        columns=alternatives_perform.index,
+    )
+
+
+def concordance_reinforced_profiles(
+    alternatives_perform: pd.DataFrame,
+    scales: Union[Dict[Any, QuantitativeScale], pd.Series],
+    weights: Union[Dict[Any, NumericValue], pd.Series],
+    indifference_thresholds: pd.DataFrame,
+    preference_thresholds: pd.DataFrame,
+    reinforced_thresholds: pd.DataFrame,
+    reinforcement_factors: Union[Dict[Any, NumericValue], pd.Series],
+    profiles_perform: pd.DataFrame,
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """_summary_
+
+    :param alternatives_perform: _description_
+    :param scales: _description_
+    :param weights: _description_
+    :param indifference_thresholds: _description_
+    :param preference_thresholds: _description_
+    :param reinforced_thresholds: _description_
+    :param reinforcement_factors: _description_
+    :param profiles_perform: _description_
+
+    :return: _description_
+    """
+    return pd.DataFrame(
+        [
+            [
+                concordance_reinforced_comprehensive(
+                    alternatives_perform.loc[alt_name],
+                    profiles_perform.loc[prof_name],
+                    scales,
+                    weights,
+                    indifference_thresholds.loc[prof_name],
+                    preference_thresholds.loc[prof_name],
+                    reinforced_thresholds.loc[prof_name],
+                    reinforcement_factors,
+                )
+                for prof_name in profiles_perform.index
+            ]
+            for alt_name in alternatives_perform.index
+        ],
+        index=alternatives_perform.index,
+        columns=profiles_perform.index,
+    ), pd.DataFrame(
+        [
+            [
+                concordance_reinforced_comprehensive(
+                    profiles_perform.loc[prof_name],
+                    alternatives_perform.loc[alt_name],
+                    scales,
+                    weights,
+                    indifference_thresholds.loc[prof_name],
+                    preference_thresholds.loc[prof_name],
+                    reinforced_thresholds.loc[prof_name],
+                    reinforcement_factors,
+                )
+                for alt_name in alternatives_perform.index
+            ]
+            for prof_name in profiles_perform.index
+        ],
+        index=profiles_perform.index,
+        columns=alternatives_perform.index,
+    )
+
+
+class Interaction:
+    def __init__(self, interaction_type: str = None,
+                 function: str = None,
+                 factor: NumericValue = None):
+        self.interaction = interaction_type
+        self.function = function
+        self.factor = factor
+
+    def set_interaction(self, interaction_type: str,
+                        function: str,
+                        factor: NumericValue):
+        self.interaction = interaction_type.upper()
+        self.function = function
+        self.factor = factor

```

Comment:
why there's a setter here? and why interaction string is not changing to uppercase in the constructor?